client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,68 +14,62 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
-
-	getInitialState() {
-		return {
-			value: '',
-			isError: false
-		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddMedia: noop,
-			onClose: noop
-		};
-	},
-
-	upload( event ) {
-		var isError = ! event.target.checkValidity();
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-		this.setState( {
-			isError: isError
-		} );
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
+
+	state = {
+		value: '',
+		isError: false,
+	};
+
+	upload = event => {
+		const isError = ! event.target.checkValidity();
+
+		this.setState( { isError } );
 
 		if ( isError || ! this.props.site ) {
 			return;
 		}
 
+		event.preventDefault();
+
 		MediaActions.clearValidationErrors( this.props.site.ID );
 		MediaActions.add( this.props.site.ID, this.state.value );
 
-		this.replaceState( this.getInitialState() );
+		this.setState( this.getInitialState() );
```

Comment:
Careful with a const, it can still get mutated, but the ref can't update. eg `initialState.value ='bar'`